seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/catalog/PaimonCatalog.java

```diff
@@ -161,6 +163,22 @@ public void createTable(TablePath tablePath, CatalogTable table, boolean ignoreI
             throw new TableAlreadyExistException(this.catalogName, tablePath);
         } catch (org.apache.paimon.catalog.Catalog.DatabaseNotExistException e) {
             throw new DatabaseNotExistException(this.catalogName, tablePath.getDatabaseName());
+        } catch (Exception e) {
+            if (e.getCause() instanceof UnsupportedOperationException) {
```

Comment:
If there throw a UnsupportedOperationException, it's must be a Paimon primary key datatype is unsupported?

Reply:
Added pattern matching